ingestion/src/metadata/ingestion/source/database/dbt/metadata.py

```diff
@@ -339,10 +339,11 @@ def yield_data_models(
                 )
             for key, manifest_node in manifest_entities.items():
                 try:
+                    resource_type = getattr(manifest_node.resource_type, "value", manifest_node.resource_type)
                     # If the run_results file is passed then only DBT tests will be processed
                     if (
                         dbt_objects.dbt_run_results
-                        and manifest_node.resource_type.value
+                        and resource_type
```

Comment:
What are the cases where `resource_type` would be `None` since we have `resource_type = getattr(manifest_node.resource_type, "value", manifest_node.resource_type)` on ln 342?

Reply:
@TeddyCr the `resource_type` is a required attribute for the dbt nodes so it'll always be there

Reply:
We should be ok removing this check then right @OnkarVO7. cc @aballiet

Reply:
Yes, will do it 👍

Reply:
No in fact it's an equality check on `resource_type`